app/models/base.py

```diff
@@ -40,3 +45,54 @@ class Config:
         match their type annotations"""
 
         validate_assignment = True
+
+
+class BaseResponse(TableModelMixin):
+    pass
+
+
+class BaseRequest(BaseModel):
+    class Meta:
+        related_fields = {}
+        model = None
+
+    def get_model(self):
+        if not self.Meta.model:
+            raise NotImplementedError(
+                "Either set the Meta.model property or override the get_model() method."
+            )
+        return self.Meta.model
+
+    def get_related_fields(self):
+        return self.Meta.related_fields
+
+    def translate(self):
```

Comment:
Likewise would it be be possible have some type hints here, just so I know what I am expecting to get back.